src/jesse_state.erl

```diff
@@ -61,7 +62,7 @@
                                      ) -> list() | no_return()
                                             )
          , default_schema_ver :: binary()
-         , schema_loader_fun  :: fun(( binary()
+         , schema_loader_fun  :: fun(( string()
```

Comment:
why this change here ? can we maintain compatibility ?
there are already a lot of match clauses depending on binary